### ISSUE TITLE
Change the first layer of seq2seq model to a merged layer

### ISSUE THREAD

**Issue by shaifugpt** [keras-team/keras#8464]
I have to change the first layer of seq2seq model to a merged layer. 

Error: list indices must be integers, not list

I am doing as:
encoder_inputs = Input(shape=(None, input_dim))
left=LSTM(64,activation='sigmoid',return_state=True,input_shape=(None,input_dim))(encoder_inputs)
right=LSTM(64,activation='sigmoid',return_state=True,go_backwards=True,input_shape=(None,input_dim))(encoder_inputs)
encoder =Concatenate([left, right])
encoder_outputs, state_h, state_c = encoder([encoder_inputs,encoder_inputs])[#error in this line]

**Comment by Dref360** [comment]
https://keras.io/layers/merge/#concatenate

**Comment by vianamp** [comment]
I think because you have `return_state=True`, the LSTM layer is returning the states in addition to the output you expect. Try to replace 

`left = LSTM(64,activation='sigmoid',return_state=True,input_shape=(None,input_dim))(encoder_inputs)`

and

`right = LSTM(64,activation='sigmoid',return_state=True,go_backwards=True,input_shape=(None,input_dim))(encoder_inputs)`

with

`left, _ , _ = LSTM(64,activation='sigmoid',return_state=True,input_shape=(None,input_dim))(encoder_inputs)`

and

`right, _ , _ = LSTM(64,activation='sigmoid',return_state=True,go_backwards=True,input_shape=(None,input_dim))(encoder_inputs)`

-m

**Comment by shaifugpt** [comment]
@vianamp I tried, but still the same error

**Comment by vianamp** [comment]
What if you leave your code as is but with `return_state=False`?

**Comment by shaifugpt** [comment]
@vianamp Still the same error after return_state=False

**Comment by vianamp** [comment]
Hi. Can you post your full script?
